bindgen/src/bound-model.ts

```diff
@@ -378,7 +401,23 @@ export class Field {
 export class Struct extends NamedType {
   readonly kind = "Struct";
   cppName!: string;
-  fields: Field[] = [];
+  private _fields: { [name: string]: Field } = {};
+
+  get fields(): Readonly<Field[]> {
+    return Object.values(this._fields);
+  }
+
+  addField(field: Field) {
+    assert(!(field.name in this._fields), `Duplicate field name on record/struct '${this.name}': '${field.name}'.`);
```

Comment:
I'm pretty sure it is impossible[1] to express duplicate fields in the spec file syntax right now, but I suppose it is fine to leave it incase that changes. 

[1] You _could_ have duplicates in the spec.yaml file, but the yaml spec [requires](https://yaml.org/spec/1.2.2/#mapping) that keys in a mapping are unique, so such a file isn't valid yaml. And even if the yaml parser doesn't detect that, since it parses into a Plain Old JS Object, any later field will completely replace the prior duplicate, so by the time you get to this code, you won't have any trace of the in-file duplication.

Reply:
Good point!